ddtrace/propagation/http.py

```diff
@@ -550,35 +554,29 @@ def _get_traceparent_values(tp):
         Otherwise we extract the trace-id, span-id, and sampling priority from the
         traceparent header.
         """
+        tp = tp.strip()
+        if not re.match(_TRACEPARENT_HEX_REGEX, tp) or tp.startswith("ff"):
+            # ff is an invalid traceparent version: https://www.w3.org/TR/trace-context/#version
+            raise ValueError("W3C traceparent hex is invalid: %s" % tp)
 
         version, trace_id_hex, span_id_hex, trace_flags_hex = tp.strip().split("-")
-        # check version is a valid hexadecimal, if not it's invalid we will move on to the next prop method
-        int(version, 16)
-        # https://www.w3.org/TR/trace-context/#version
-        if version == "ff":
-            raise ValueError("'ff' is an invalid traceparent version")
-        # currently 00 is the only version format, but if future versions come up we may need to add changes
         if version != "00":
```

Comment:
Removing the comment? it seems good to assert that this should be `00`, but we don't error if it isn't.

Reply:
The comment was move to the function that validates the regex